### PR TITLE
Check values while apply settings

### PR DESCRIPTION
**Pull request by Chaoba**
### Summary of changes
In this PR

- Add checks while applySettings to plugins. Only apply setting when set a new value
- Add tests for all values of enums.

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `./gradlew apiDump` to update generated api files, if there's public API changes, otherwise the `verify-kotlin-binary-compatibility` CI will fail.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Review by alexander-kulikovskii — APPROVED** [review]

**Comment by yunikkk** [comment]
@Chaoba btw does this fix some specific issue or is just an optimisation to avoid unnecessary changes?

**Comment by Chaoba** [comment]
> @Chaoba btw does this fix some specific issue or is just an optimisation to avoid unnecessary changes?

Just an optimisation

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 Mapbox welcomes participation and contributions from everyone.
 
 # main
+## Features ✨ and improvements 🏁
+* Add checks while applySettings to plugins. Only apply setting when setting a new value. ([#1189](https://github.com/mapbox/mapbox-maps-android/pull/1189))
```
Comment (yunikkk) [inline CHANGELOG.md:7]: Maybe rephrase to highlight the main idea of the change, e.g.
```
* Optimize how plugins handle settings changes. Call `applySettings` only when settings value changes. 
```
?

**Review by yunikkk — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Chaoba) [reply CHANGELOG.md:7]: Sounds good.
